### ISSUE TITLE
In SSR mode, when the page uses the <a href="#hash">hash<a> aiming point, it cannot be accurately positioned.

### ISSUE THREAD

**Issue by daxiongYang** [quasarframework/quasar#10970]
**Describe the bug**
When a content list needs to be added to the page, the A-tag aiming point cannot be accurately located.

**Codepen/jsFiddle/Codesandbox (required)**
Sorry, it is not possible to show the effect through the above method. 

**To Reproduce**
1. Visit the official website https://quasar.dev/start/quasar-cli
2. Execute the following code through the browser console:
  ```js
    // Add an A tag with an aiming point to the content of the document
    let eleA = document.createElement('a');
    eleA.href = '#hash';
    eleA.innerText = 'hash';
    let domDoc = document.querySelector('.doc-page');
    domDoc.insertBefore(eleA, domDoc.firstChild);
  ```
3. Click on the added A tag.
4. Pay attention to the URL(https://quasar.dev/#hash) It should not be like this! !

**Expected behavior**
URL = https://quasar.dev/start/quasar-cli#hash

**Platform (please complete the following information):**
Quasar Version: "^2.0.0",
@quasar/app Version: "^3.0.0",

Quasar mode:
  - [ ] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [ ] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: v11.0.1
Node: v14.18.0
NPM: v7.24.1
Yarn:
Browsers: v94.0.4606.81
iOS:
Android:
Electron:

**Comment by rstoenescu** [comment]
Clicking on a link with href="#something" and appending the "#something" part to the current browser URL is a native browser behavior.
This ticket has nothing to do with Quasar.

**Comment by daxiongYang** [comment]
@rstoenescu Thank you very much for your efficient response.
I know this is the behavior of the browser, but I really don't understand why this result occurs! I have searched a lot of information and still can't solve this problem! !
Maybe I only explained one `#hash` situation above. In fact, the relative path including `../path` `./path` has become invalid! !
for example:
`https://quasar.dev/start/quasar-cli` Jump to `https://quasar.dev/start/test` Still invalid!
```js
    let eleA = document.createElement('a');
    eleA.href = '../test';
    eleA.innerText = 'Test';
    let domDoc = document.querySelector('.doc-page');
    domDoc.insertBefore(eleA, domDoc.firstChild);
```
